src/main/java/se/sundsvall/contract/integration/db/specification/ContractSpecification.java

```diff
@@ -27,20 +29,27 @@ public static Specification<ContractEntity> createContractSpecification(final St
 
 			predicates.add(criteriaBuilder.equal(root.get(MUNICIPALITY_ID), municipalityId));
 
-			if (request.landLeaseType() != null) {
+			if (StringUtils.isNotBlank(request.landLeaseType())) {
 				predicates.add(criteriaBuilder.equal(root.get(LAND_LEASE_TYPE), request.landLeaseType()));
 			}
 
-			if (request.end() != null) {
+			if (StringUtils.isNotBlank(request.end())) {
 				predicates.add(criteriaBuilder.equal(root.get(END), LocalDate.parse(request.end())));
 			}
 
-			if (request.externalReferenceId() != null) {
+			if (StringUtils.isNotBlank(request.externalReferenceId())) {
 				predicates.add(criteriaBuilder.equal(root.get(EXTERNAL_REFERENCE_ID), request.externalReferenceId()));
 			}
 
-			if (request.propertyDesignation() != null) {
-				predicates.add(criteriaBuilder.equal(root.get(PROPERTY_DESIGNATION), request.propertyDesignation()));
+			if(!CollectionUtils.isEmpty(request.propertyDesignations())) {
+
+				final List<Predicate> propertyPredicate = new ArrayList<>();
+
+				for(String property : request.propertyDesignations()) {
+					propertyPredicate.add(criteriaBuilder.isMember(property, root.get(PROPERTY_DESIGNATIONS)));
+				}
+
+				predicates.add(criteriaBuilder.or(propertyPredicate.toArray(new Predicate[0])));
```

Comment:
Maybe check if `propertyPredicate` is empty or not before adding it/them to `predicates`?